In [32]:
import pandas as pd

#### Import data (see *minwage.py*)

In [33]:
df_raw = pd.read_csv(r'./minwage_raw.csv',sep=',')
print(df_raw.to_string())

   State or otherjurisdiction             1968 (a)             1970 (a)               1972        1976 (a)            1979            1980            1981            1988            1991            1992            1994            1996            1997            1998              2000              2001              2002              2003              2004              2005             2006            2007          2008          2009              2010              2011              2012              2013          2014          2015          2016          2017          2018           2019          2020          2021          2022          2023
0              Federal (FLSA)        $1.15 & $1.60        $1.30 & $1.60              $1.60   $2.20 & $2.30           $2.90           $3.10           $3.35           $3.35           $3.80           $4.25           $4.25           $4.25           $4.75           $5.15             $5.15             $5.15             $5.15             $5.15             

#### Drop non-relevant obs

In [34]:
df_raw = df_raw.drop(['1968 (a)','1970 (a)','1972','1976 (a)','1979','1980','1981','1988'],axis=1)
df_raw = df_raw[df_raw['State or otherjurisdiction'] != 'Guam']
df_raw = df_raw[df_raw['State or otherjurisdiction'] != 'Puerto Rico']
df_raw = df_raw[df_raw['State or otherjurisdiction'] != 'U.S. Virgin Islands']
df_raw = df_raw.replace({'\$':''}, regex=True)

In [35]:
df = df_raw

#### Decisions

> from: https://www.debt.org/jobs/minimum-wage/#:~:text=What%20Happens%20if%20a%20State's,they%20can%20be%20paid%20less.
>
> **Do States Have to Follow the Federal Minimum Wage?**
>
>States are required to follow federal minimum wage law. States can pass their own laws to make the wage higher, equal to or lower than the federal law, but they can’t make >other changes that overrule the federal law, for instance, who is exempt or how many hours constitutes a work week.
>
>Federal minimum wage overrides state if the state wage is lower. It doesn’t if the state wage is higher. Hourly workers who come under Fair Labor Standard Act guidelines, which >most hourly workers do, always get the higher wage. Some cities have higher minimum wage laws than their state wage. In those cases, too, the higher wage prevails.
>
>If a state has no minimum wage law, the federal wage prevails.

Apply effective minimum wage

In [36]:
for i in range(len(df.iloc[:,0])):
    for j in range(len(df.columns)):
        if df.iloc[i,j] == '...':
            df.iloc[i,j] = df.iloc[0,j]

>[c] - Rates applicable to employers of four or more.
>
>(d) - Rates applicable to employers of six or more. In West Virginia, applicable to employers of six or more in one location.
>
>(e) Rates applicable to employers of two or more.

Select higher minimum wage

In [37]:
df = df.replace({'\[c\]':''}, regex=True).replace({'\(d\)':''}, regex=True).replace({'\(e\)':''}, regex=True)

> (g) - Minnesota sets a lower rate for enterprises with annual receipts of less than $500,000 ($4.90, January 1, 1998-January 1, 2005). The dollar amount prior to September 1, 1997 was $362,500 ($4.00 - January 1, 1991-January 1, 1997); Montana sets a lower rate for businesses with gross annual sales of $110,000 or less ($4.00 - January 1, 1992-January 1, 2005); Ohio sets a lower rate for employers with gross annual sales from $150,000 to $500,000 ($3.35 - January 1, 1991-January 1, 2005) and for employers with gross annual sales under $150,000 ($2.50 - January 1, 1991-January 1, 2005); Oklahoma sets a lower rate for employers of fewer than 10 full-time employees at any one location and for those with annual gross sales of less than $100,000 ($2.00, January 1, 1991-January 1, 2005); and the U.S. Virgin Islands sets a lower rate for businesses with gross annual receipts of less than $150,000 ($4.30, January 1, 1991-January 1, 2005).
>
> (h) - In the District of Columbia, wage orders were replaced by a statutory minimum wage on October 1, 1993. A $5.45 minimum rate remained in effect for the laundry and dry cleaning industry as the result of the grandfather clause.

Select higher minimum wage

In [38]:
df = df.replace({'\(g\)':''}, regex=True).replace({'\(h\)':''}, regex=True)

There are States with two different minimum wages (not clarified, maybe FLSA (?))

In [39]:
for i in range(len(df.iloc[:,0])):
    for j in range(len(df.columns)):
        if '-' in df.iloc[i,j]:
            df.iloc[i,j] = df.iloc[i,j].split('-')[1].lstrip().split(' ')[0]
        elif '/' in df.iloc[i,j]:
            df.iloc[i,j] = df.iloc[i,j].split('/')[1].lstrip().split(' ')[0]

#### Remove blank spaces, strings to floats

In [44]:
df = df.replace({' ':''}, regex=True)
df.iloc[:,1:-1] = df.iloc[:,1:-1].astype(float)
df = df.melt(id_vars="State or otherjurisdiction", 
        var_name="Year", 
        value_name="Value")
print(df.to_string())

     State or otherjurisdiction  Year  Value
0                 Federal(FLSA)  1991    3.8
1                       Alabama  1991    3.8
2                        Alaska  1991    4.3
3                       Arizona  1991    3.8
4                      Arkansas  1991   3.35
5                    California  1991   4.25
6                      Colorado  1991    3.0
7                   Connecticut  1991   4.25
8                      Delaware  1991    3.8
9                       Florida  1991    3.8
10                      Georgia  1991   3.25
11                       Hawaii  1991   3.85
12                        Idaho  1991    3.8
13                     Illinois  1991    3.8
14                      Indiana  1991   3.35
15                         Iowa  1991   4.25
16                       Kansas  1991   2.65
17                     Kentucky  1991    3.8
18                    Louisiana  1991    3.8
19                        Maine  1991   3.85
20                     Maryland  1991    3.8
21        

In [45]:
df.to_csv(r'./minwage_clean_ub.csv',sep=',',index=False)

#### Notice:

>from: https://www.debt.org/jobs/minimum-wage/#:~:text=What%20Happens%20if%20a%20State's,they%20can%20be%20paid%20less.
>
> **What Happens if a State’s Minimum Wage Is Lower Than the Federal Minimum Wage?**
>
>In places where the state minimum wage is lower than the federal wage, workers who come under the Fair Labor Standards Act earn the federal wage.
>
>If workers don’t come under the FLSA, they can be paid less. Georgia, Oklahoma and Wyoming have minimum wages lower than the federal wage.
>
>Georgia and Wyoming’s are both $5.15 an hour. In Oklahoma employers pay the federal minimum wage if their business is a certain size that’s more generous than federal requirements. If not, workers get $2 an hour.

>from: https://www.dol.gov/agencies/whd/compliance-assistance/handy-reference-guide-flsa
>
>All employees of certain enterprises having workers engaged in interstate commerce, producing goods for interstate commerce, or handling, selling, or otherwise working on goods or materials that have been moved in or produced for such commerce by any person, are covered by the **FLSA**.
>
>A covered enterprise is the related activities performed through unified operation or common control by any person or persons for a common business purpose and —
>
>>whose annual gross volume of sales made or business done is not less than $500,000 (exclusive of excise taxes at the retail level that are separately stated); or
>>
>> is engaged in the operation of a hospital, an institution primarily engaged in the care of the sick, the aged, or the mentally ill who reside on the premises; a school for mentally or physically disabled or gifted children; a preschool, an elementary or secondary school, or an institution of higher education (whether operated for profit or not for profit); or
>>
>>is an activity of a public agency.
Any enterprise that was covered by the FLSA on March 31, 1990, and that ceased to be covered because of the revised $500,000 test, continues to be subject to the overtime pay, child labor and recordkeeping provisions of the FLSA.
>